src/methods/run-function.ts

```diff
@@ -14,112 +13,94 @@ export interface RunFunctionResult {
+  functionRunnerPath: string,
+  wasmPath: string,
+  fixture: FixtureData,
+  queryPath: string,
+  schemaPath: string,
 ): Promise<RunFunctionResult> {
   try {
-    const inputJson = JSON.stringify(input);
-
-    let functionDir, appRootDir, functionName;
-    
-    if (functionPath !== undefined && functionPath !== null) {
-      // Use provided function path
-      functionDir = path.resolve(functionPath);
-      appRootDir = path.dirname(functionDir);
-      functionName = path.basename(functionDir);
-    } else {
-      // Calculate paths correctly for when used as a dependency:
-      // __dirname = /path/to/function/tests/node_modules/function-testing-helpers/src/methods
-      // Go up 5 levels to get to function directory: ../../../../../ = /path/to/function
-      functionDir = path.dirname(path.dirname(path.dirname(path.dirname(path.dirname(__dirname)))));
-      appRootDir = path.dirname(functionDir);
-      functionName = path.basename(functionDir);
-    }
-    
-    return new Promise((resolve, reject) => {
-      const shopifyProcess = spawn('shopify', [
-        'app', 'function', 'run',
-        '--export', exportName,
+    const inputJson = JSON.stringify(fixture.input);
+
+    return new Promise((resolve) => {
+      const runnerProcess = spawn(functionRunnerPath, [
+        '-f', wasmPath,
+        '--export', fixture.export,
+        '--query-path', queryPath,
+        '--schema-path', schemaPath,
         '--json',
-        '--path', functionName
       ], {
-        cwd: appRootDir,
-        stdio: ['pipe', 'pipe', 'pipe']
+        stdio: ['pipe', 'pipe', 'pipe'],
       });
 
       let stdout = '';
       let stderr = '';
 
-      shopifyProcess.stdout.on('data', (data) => {
+      runnerProcess.stdout.on('data', (data) => {
         stdout += data.toString();
       });
 
-      shopifyProcess.stderr.on('data', (data) => {
+      runnerProcess.stderr.on('data', (data) => {
         stderr += data.toString();
       });
 
-      shopifyProcess.on('close', (code) => {
+      runnerProcess.on('close', (code) => {
+
         if (code !== 0) {
           resolve({
             result: null,
-            error: `Command failed with exit code ${code}: ${stderr}`,
+            error: `function-runner failed with exit code ${code}: ${stderr}`
           });
           return;
         }
 
-        let result;
         try {
-          result = JSON.parse(stdout);
-
-          let actualOutput;
-          if (result?.output?.humanized) {
-            actualOutput = JSON.parse(result.output.humanized);
-          } else if (result?.output) {
-            actualOutput = result.output;
-          } else {
-            actualOutput = result;
-          }
+          const result = JSON.parse(stdout);
 
+          // function-runner output format: { output: {...} }
           resolve({
-            result: { output: actualOutput },
-            error: null,
+            result: { output: result.output || result },
```

Comment:
Why do we need both of both of these? Is the output from function-runner not consistent?

I just did some quick testing and I can see that if we pass in invalid json as the input, we don't get the "normal" formatted response with a `result { "output": {}`, but just `Error: Invalid input JSON: trailing characters at line 17 column 2`. Is this what we're covering here? Wouldn't that result in a parse error and end up in the `catch`?